hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/OpensslSm4CtrCryptoCodec.java

```diff
@@ -41,6 +41,10 @@ public OpensslSm4CtrCryptoCodec() {
     if (loadingFailureReason != null) {
       throw new RuntimeException(loadingFailureReason);
     }
+
+    if (!OpensslCipher.isSupported(CipherSuite.SM4_CTR_NOPADDING)) {
+      throw new RuntimeException("Doesn't support SM4 CTR.");
```

Comment:
How about making the message "the openssl native library is built without SM4 CTR support"?

the rest looks good to me.

Reply:
Sure. I update the message as you suggest.